.github/workflows/drivers.yml

```diff
@@ -12,8 +12,48 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  # This job will determine if the tier 3 driver tests should be skipped.
+  # It will always run first.
+  determine-skip-tier3-drivers:
+    runs-on: ubuntu-latest
+    outputs:
+      skip_tier3: ${{ steps.check_team_and_branch.outputs.skip }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Read team.json and check if tier3 drivers should run
+        id: check_team_and_branch
+        run: |
+          GITHUB_ACTOR_LOWER=$(echo "${{ github.actor }}" | tr '[:upper:]' '[:lower:]')
+          REQUIRED_TEAM_MEMBERS=$(jq -r '.teams[] | select(.name == "Drivers" or .name == "Querying" or .name == "Semantic Layer") | .members[]' .github/team.json | tr '[:upper:]' '[:lower:]')
+          IS_REQUIRED_MEMBER=false
+          for member in ${REQUIRED_TEAM_MEMBERS}; do
+            if [[ "${GITHUB_ACTOR_LOWER}" == "${member}" ]]; then
+              IS_REQUIRED_MEMBER=true
+              break
+            fi
+          done
```

Comment:
since we already set labels based on team membership, can we just rely on the team labels?

Reply:
Hrm, I was worried about the order of things here, if I open a pr then set myself to assignee, which adds the team Driver label, would that rerun the skipped tests?

Reply:
Went with labels, we can go back to teams.json check if it proves to be flaky.